src/Discord/Parts/Channel/Poll/PollAnswer.php

```diff
@@ -0,0 +1,172 @@
+ */
+class PollAnswer extends Part
+{
+    /**
+     * {@inheritdoc}
+     */
+    protected $fillable = [
+        'answer_id',
+        'poll_media',
+
+        // events
+        'user_id',
+        'channel_id',
+        'message_id',
+        'guild_id',
+    ];
+
+    /**
+     * Gets the user attribute.
+     *
+     * @return User
+     */
+    protected function getUserAttribute(): User
+    {
+        return $this->discord->users->get('id', $this->user_id);
+    }
+
+    /**
+     * Gets the channel attribute.
+     *
+     * @return Channel|Thread
+     */
+    protected function getChannelAttribute(): Part
+    {
+        if ($guild = $this->guild) {
+            $channels = $guild->channels;
+
+            if ($channel = $channels->get('id', $this->channel_id)) {
+                return $channel;
+            }
+
+            foreach ($channels as $parent) {
+                if ($thread = $parent->threads->get('id', $this->channel_id)) {
+                    return $thread;
+                }
+            }
+        }
+
+        // @todo potentially slow
+        if ($channel = $this->discord->getChannel($this->channel_id)) {
+            return $channel;
+        }
+
+        return $this->factory->part(Channel::class, [
+            'id' => $this->channel_id,
+            'type' => Channel::TYPE_DM,
+        ], true);
+    }
+
+    /**
+     * Gets the message attribute.
+     *
+     * @return Message|null
+     */
+    protected function getMessageAttribute(): ?Message
+    {
+        if ($channel = $this->channel) {
+            return $channel->messages->get('id', $this->message_id);
+        }
+
+        return $this->attributes['message'] ?? null;
+    }
+
+    /**
+     * Returns the guild that owns the channel the message was sent in.
+     *
+     * @return Guild|null
+     */
+    protected function getGuildAttribute(): ?Guild
+    {
+        if (! isset($this->guild_id)) {
+            return null;
+        }
+
+        return $this->discord->guilds->get('id', $this->guild_id);
+    }
+
+    /**
+     * Returns the users that voted for the answer.
+     *
+     * @param int       $limit  The maximum number of users to return.
+     * @param int|null  $after  The user ID to get users after.
+     *
+     * @link https://discord.com/developers/docs/resources/poll#get-answer-voters
+     *
+     * @throws \OutOfRangeException
+     *
+     * @return ExtendedPromiseInterface<Collection|User[]>
+     */
+    public function getVoters(int $limit = 25, ?int $after = null): ExtendedPromiseInterface
+    {
+        if ($limit < 1 || $limit > 100) {
+            throw new \OutOfRangeException('Limit must be between 1 and 100.');
+        }
+
+        $endpoint = Endpoint::bind(Endpoint::CHANNEL_POLL_ANSWERS, $this->channel_id, $this->message_id, $this->answer_id);
+
+        $endpoint->addQuery('limit', $limit);
+
+        if ($after) {
+            $endpoint->addQuery('after', $after);
+        }
```

Comment:
I didn't notice the `OptionsResolver` (e.g. in [reactions](https://github.com/discord-php/DiscordPHP/blob/master/src/Discord/Parts/Channel/Reaction.php#L148-L154)) so this could probably get changed to be uniform.